ci_scripts/sync_target_database.py

```diff
@@ -45,31 +50,55 @@ def save_target_database(target_database_text: str) -> Path:
     Returns:
         The path to targets.json.
     """
-    output_file_path = Path(
-        configuration.get_value(ConfigurationVariable.PROJECT_ROOT), "mbed_targets", "data", "targets.json"
-    )
     output_file_path.parent.mkdir(exist_ok=True)
     output_file_path.write_text(target_database_text)
-    return output_file_path
 
 
-def git_commit_and_push(file_to_commit: Path, branch_name: str, commit_msg: str) -> None:
+def diff_objects_from_json_str(incoming_text: str, current_text: str) -> List[dict]:
+    """Convert two json strings to objects and take the difference."""
+    current = json.loads(current_text)
+    incoming = json.loads(incoming_text)
+    return [i for i in current + incoming if i not in current or i not in incoming]
+
+
+def create_new_boards_news_item_text(new_targets: dict) -> str:
+    """Create a new news item string from the list of new targets."""
+    item_str = "New boards added:"
+    board_names = ", ".join(target["attributes"]["board_type"] for target in new_targets)
+    return f"{item_str} {board_names}"
+
+
+def write_news_file(item_text: str, file_name: str) -> Path:
+    """Write some text to a news file in the news directory of the repository."""
+    news_file_path = Path(configuration.get_value(ConfigurationVariable.PROJECT_ROOT), "news", f"{file_name}.misc")
```

Comment:
```suggestion
    news_file_path = Path(configuration.get_value(ConfigurationVariable.PROJECT_ROOT), "news", f"{file_name}.feature")
```
This is to actually change the version of the package and have a record in the changelog. misc is only mentioned and does not change the version number